src/micro_http/src/server.rs

```diff
@@ -379,6 +379,21 @@ impl HttpServer {
         Ok(parsed_requests)
     }
 
+    /// This function is responsible with flushing any remaining outgoing
+    /// requests on the server.
+    ///
+    /// Note that this function can block the thread on write, since the
+    /// operation is blocking.
+    pub fn flush_outgoing_writes(&mut self) {
+        for (_, connection) in self.connections.iter_mut() {
+            while connection.state == ClientConnectionState::AwaitingOutgoing {
+                if connection.write().is_err() {
```

Comment:
I have added this error handling for the case in which an error such as a ConnectionClosed would not allow the function to set the proper ClientConnectionState. In that case, we want to break out of the loop.

Reply:
Please also log a warning/error here.

Reply:
Done